include/boost/url/decoded_view.hpp

```diff
@@ -0,0 +1,1049 @@
+        string_view lhs,
+        decoded_view rhs) noexcept
+    {
+        return rhs.compare(lhs) > 0;
+    }
+
+    /** Compares two strings
+
+        @param lhs A string to compare
+        @param rhs A string to compare
+
+        @return true if the corresponding comparison holds after percent decoding
+     */
+    friend BOOST_CXX14_CONSTEXPR
+    bool
+    operator<=(
+        decoded_view lhs,
+        decoded_view rhs) noexcept
+    {
+        return lhs.compare(rhs) <= 0;
+    }
+
+    /// @copydoc operator<=
+    friend BOOST_CXX14_CONSTEXPR
+    bool
+    operator<=(
+        decoded_view lhs,
+        string_view rhs) noexcept
+    {
+        return lhs.compare(rhs) <= 0;
+    }
+
+    /// @copydoc operator<=
+    friend BOOST_CXX14_CONSTEXPR
+    bool
+    operator<=(
+        string_view lhs,
+        decoded_view rhs) noexcept
+    {
+        return rhs.compare(lhs) >= 0;
+    }
+
+    /** Compares two strings
+
+        @param lhs A string to compare
+        @param rhs A string to compare
+
+        @return true if the corresponding comparison holds after percent decoding
+     */
+    friend BOOST_CXX14_CONSTEXPR
+    bool
+    operator>(
+        decoded_view lhs,
+        decoded_view rhs) noexcept
+    {
+        return lhs.compare(rhs) > 0;
+    }
+
+    /// @copydoc operator>
+    friend BOOST_CXX14_CONSTEXPR
+    bool
+    operator>(
+        decoded_view lhs,
+        string_view rhs) noexcept
+    {
+        return lhs.compare(rhs) > 0;
+    }
+
+    /// @copydoc operator>
+    friend BOOST_CXX14_CONSTEXPR
+    bool
+    operator>(
+        string_view lhs,
+        decoded_view rhs) noexcept
+    {
+        return rhs.compare(lhs) < 0;
+    }
+
+    /** Compares two strings
+
+        @param lhs A string to compare
+        @param rhs A string to compare
+
+        @return true if the corresponding comparison holds after percent decoding
+     */
+    friend BOOST_CXX14_CONSTEXPR
+    bool
+    operator>=(
+        decoded_view lhs,
+        decoded_view rhs) noexcept
+    {
+        return lhs.compare(rhs) >= 0;
+    }
+
+    /// @copydoc operator>=
+    friend BOOST_CXX14_CONSTEXPR
+    bool
+    operator>=(
+        decoded_view lhs,
+        string_view rhs) noexcept
+    {
+        return lhs.compare(rhs) >= 0;
+    }
+
+    /// @copydoc operator>=
+    friend BOOST_CXX14_CONSTEXPR
+    bool
+    operator>=(
+        string_view lhs,
+        decoded_view rhs) noexcept
+    {
+        return rhs.compare(lhs) <= 0;
+    }
+
+    /** Compares two strings
+
+        @note This function only participates in overload resolution
+        if we can construct a string_view from the types being
+        compared.
+
+        @tparam StringViewable A type from which we can construct a string_view
+
+        @param lhs A string to compare
+        @param rhs A StringViewable to compare
+
+        @return true if the corresponding comparison holds after percent decoding
+      */
+    template <class StringViewable>
```

Comment:
"StringViewable" doesn't look so great. How about just `String`

Reply:
I was trying to imply the string type needs to be convertible to a string view. But I guess anyone capable of writing a string type is also able to figure that out.